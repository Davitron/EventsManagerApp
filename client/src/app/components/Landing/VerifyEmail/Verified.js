import React, { Component } from 'react';
import { Row, Col } from 'react-materialize';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import UserActions from '../../../../actions/user.action';
import '../../../../App.css';

/**
 *
 */
class VerifiedEmail extends Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.state = {
      payload: null
    };
  }
  /**
   *@returns {*} makes ajax call befor render
   */
  componentWillMount() {
    const userActions = new UserActions();
    const url = new URL(window.location.href);
    const param = new URLSearchParams(url.search);
    const token = param.get('token');
    this.props.dispatch(userActions.completeRegistration(token));
  }
  /**
   *@returns {*} view for langing page
   */
  render() {
    return (
      <div className="App-main">
        <div className={['landing', 'container', 'animated', 'bounceInUp'].join(' ')}>
          <Row className='center'>
              <Col s={12} className={['light', 'white-text', 'center', 'animated', 'bounceInDown'].join(' ')}>
              <h3> <b>Your email has been veriried</b><br /></h3>
              </Col>
          </Row>
          <Row className='center'> <Link className={['btn', 'btn-large', 'waves-effect', 'blue'].join(' ')} to='/'>Continue as user</Link></Row>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    stateProps: state.userVerification
  };
};

export default connect(mapStateToProps)(VerifiedEmail);
