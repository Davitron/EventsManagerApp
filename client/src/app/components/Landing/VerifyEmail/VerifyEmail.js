import React, { Component } from 'react';
import { Row, Col, Button } from 'react-materialize';
import '../../../../App.css';

/**
 *
 */
class VerifyEmail extends Component {
  /**
   *@returns {*} view for langing page
   */
  render() {
    return (
      <div className={['landing', 'container', 'animated', 'bounceInUp'].join(' ')}>
        <Row className='center'>
          <h3>
            <Col s={12} className={['light', 'white-text', 'center', 'animated', 'bounceInDown'].join(' ')}>
              <b>A verificaton link has been sent to your email address<br />
              Check your inbox to complete your registration.</b>
            </Col>
          </h3>
        </Row>
      </div>
    );
  }
}

export default VerifyEmail;
