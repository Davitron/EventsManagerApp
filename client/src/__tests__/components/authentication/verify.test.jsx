import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import VerifyEmail from '../../../components/authentication/verify';

let mountedComponent;

let props;
const locations = [];
const history = {
  push(location) {
    locations.push(location);
  },
};

let wrapper;


/**
 * @description Initialise the component
 *
 * @returns {object} mountedComponent - Mounte
 */
const getComponent = () => {
  if (!mountedComponent) {
    props = {};
    history.push = jest.fn();
    mountedComponent = shallow(<VerifyEmail {...props} />);
  }
  return mountedComponent;
};

describe('Verified Component', () => {
  beforeEach(() => {});

  it('component successfully rendered', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
