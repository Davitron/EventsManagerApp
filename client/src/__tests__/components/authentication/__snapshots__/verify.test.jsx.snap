// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verified Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VerifyEmail />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="home"
      >
        <div
          className="section section__hero"
          style={
            Object {
              "color": "white",
            }
          }
        >
          <div
            className="my-container"
            style={
              Object {
                "paddingTop": "8em",
              }
            }
          >
            <h2
              className="animated fadeInUp"
              id="message"
            >
              Registration Successful.
            </h2>
            <h3
              className="animated fadeInUp"
              style={
                Object {
                  "paddingBottom": "60px",
                }
              }
            >
              A verification link has been sent to your mail
            </h3>
          </div>
        </div>
      </div>,
      "id": "verify",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="section section__hero"
          style={
            Object {
              "color": "white",
            }
          }
        >
          <div
            className="my-container"
            style={
              Object {
                "paddingTop": "8em",
              }
            }
          >
            <h2
              className="animated fadeInUp"
              id="message"
            >
              Registration Successful.
            </h2>
            <h3
              className="animated fadeInUp"
              style={
                Object {
                  "paddingBottom": "60px",
                }
              }
            >
              A verification link has been sent to your mail
            </h3>
          </div>
        </div>,
        "className": "home",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="my-container"
            style={
              Object {
                "paddingTop": "8em",
              }
            }
          >
            <h2
              className="animated fadeInUp"
              id="message"
            >
              Registration Successful.
            </h2>
            <h3
              className="animated fadeInUp"
              style={
                Object {
                  "paddingBottom": "60px",
                }
              }
            >
              A verification link has been sent to your mail
            </h3>
          </div>,
          "className": "section section__hero",
          "style": Object {
            "color": "white",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="animated fadeInUp"
                id="message"
              >
                Registration Successful.
              </h2>,
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                A verification link has been sent to your mail
              </h3>,
            ],
            "className": "my-container",
            "style": Object {
              "paddingTop": "8em",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Registration Successful.",
                "className": "animated fadeInUp",
                "id": "message",
              },
              "ref": null,
              "rendered": "Registration Successful.",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A verification link has been sent to your mail",
                "className": "animated fadeInUp",
                "style": Object {
                  "paddingBottom": "60px",
                },
              },
              "ref": null,
              "rendered": "A verification link has been sent to your mail",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="home"
        >
          <div
            className="section section__hero"
            style={
              Object {
                "color": "white",
              }
            }
          >
            <div
              className="my-container"
              style={
                Object {
                  "paddingTop": "8em",
                }
              }
            >
              <h2
                className="animated fadeInUp"
                id="message"
              >
                Registration Successful.
              </h2>
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                A verification link has been sent to your mail
              </h3>
            </div>
          </div>
        </div>,
        "id": "verify",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="section section__hero"
            style={
              Object {
                "color": "white",
              }
            }
          >
            <div
              className="my-container"
              style={
                Object {
                  "paddingTop": "8em",
                }
              }
            >
              <h2
                className="animated fadeInUp"
                id="message"
              >
                Registration Successful.
              </h2>
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                A verification link has been sent to your mail
              </h3>
            </div>
          </div>,
          "className": "home",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="my-container"
              style={
                Object {
                  "paddingTop": "8em",
                }
              }
            >
              <h2
                className="animated fadeInUp"
                id="message"
              >
                Registration Successful.
              </h2>
              <h3
                className="animated fadeInUp"
                style={
                  Object {
                    "paddingBottom": "60px",
                  }
                }
              >
                A verification link has been sent to your mail
              </h3>
            </div>,
            "className": "section section__hero",
            "style": Object {
              "color": "white",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2
                  className="animated fadeInUp"
                  id="message"
                >
                  Registration Successful.
                </h2>,
                <h3
                  className="animated fadeInUp"
                  style={
                    Object {
                      "paddingBottom": "60px",
                    }
                  }
                >
                  A verification link has been sent to your mail
                </h3>,
              ],
              "className": "my-container",
              "style": Object {
                "paddingTop": "8em",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Registration Successful.",
                  "className": "animated fadeInUp",
                  "id": "message",
                },
                "ref": null,
                "rendered": "Registration Successful.",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A verification link has been sent to your mail",
                  "className": "animated fadeInUp",
                  "style": Object {
                    "paddingBottom": "60px",
                  },
                },
                "ref": null,
                "rendered": "A verification link has been sent to your mail",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
